test/cpp/api/util.h

```diff
@@ -25,5 +28,16 @@ class SimpleContainer : public nn::Cloneable<SimpleContainer> {
 inline bool pointer_equal(at::Tensor first, at::Tensor second) {
   return first.data<float>() == second.data<float>();
 }
+
+inline std::string get_tempfile() {
+#ifdef WIN32
+  return std::tmpnam(nullptr);
+#else
+  // http://pubs.opengroup.org/onlinepubs/009695399/functions/mkstemp.html
+  char filename[] = "/tmp/fileXXXXXX";
+  mkstemp(filename);
```

Comment:
whoa whoa whoa, what is this? You're leaking a file descriptor here! Don't do that!

Also, an API that gives you a tempfile filename is basically *asking* for the user of the API to leak temporary files. Please clean up after yourself using an RAII class.